hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/ExitUtil.java

```diff
@@ -94,7 +94,12 @@ public String toString() {
       if (message == null) {
         message = super.toString();
       }
-      return Integer.toString(status) + ": " + message;
+      String finalString = status + ": " + message;
+      Throwable cause = getCause();
+      if (cause != null) {
+        finalString = finalString + " : \n" + "cause: " + cause.getMessage();
+      }
+      return finalString;
```

Comment:
Are we changing the response string, I am not sure about the compatibility here.
The above javadoc says
```
String value does not include exception type, just exit code and message.
```
We shouldn't make prod level API changes for MiniDFSCluster

Reply:
> just exit code and message

In a way, we are just improving the message by also including cause message if any cause is associated. I just thought it would be good improvement. We are still not including exception type even with this patch. WDYT?

Reply:
But if you still think we should not include cause.getMessage() in addition to this.getMessage(), I can remove it. It's just that we would be able to include one more layer of message to the existing message if we can include cause.getMessage() as well, just a minor improvement.